src/consensus/aft/raft.h

```diff
@@ -3111,10 +3111,15 @@ namespace aft
       leader_id.reset();
 
       CCF_ASSERT_FMT(
-        !retirement_idx.has_value(),
-        "retirement_idx already set to {}",
-        retirement_idx.value());
-      retirement_idx = state->commit_idx;
+        !retirement_idx.has_value() ||
+          retirement_idx.value() <= state->commit_idx,
+        "unexpected attempt to pull retirement_idx forward from {} to {}",
+        retirement_idx.value(),
+        state->commit_idx);
+      if (!retirement_idx.has_value())
```

Comment:
That does not seem right to me, the only time retirement_idx can change is if a retirement gets rolled back. Unless I'm missing something retirement_idx is being given different semantics in 2tx, compared to 1tx, and I don't think we want to go that way.

Reply:
We don't, that's right! Perhaps we should leave retirement_idx entirely unassigned until entering `RETIRED`, instead of `RETIRING`.